### PR TITLE
Merge selected airlift changes, round four

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: http-server/src/main/java/com/proofpoint/http/server/HackGzipHandler.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.eclipse.jetty.server.handler.gzip.GzipHandler;
+
+import java.util.zip.Deflater;
+
+// TODO: remove this when fixed in Jetty
+class HackGzipHandler
+        extends GzipHandler
+{
+    @Override
+    public Deflater getDeflater(Request request, long content_length)
```
Comment (gwittel) [inline http-server/src/main/java/com/proofpoint/http/server/HackGzipHandler.java]: Looks like this was fixed back in April (eclipse/jetty.project/issues/491)  Do we still need this?  Or will this be removed in a later PR?

Comment (johngmyers) [inline http-server/src/main/java/com/proofpoint/http/server/HackGzipHandler.java]: Looks like. I'll try sending airlift a pull request.

**Comment by gwittel** [comment]
LGTM

**Comment by johngmyers** [comment]
This is on hold pending resolution of Jetty problems. Closing for now.

**Pull request closed without merging**